plugins/filter/lib/filter.cpp

```diff
@@ -0,0 +1,208 @@
+/*
+ * This file is a part of the Voice Call Manager project
+ *
+ * Copyright (C) 2024  Damien Caliste <dcaliste@free.fr>
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License along
+ * with this program; if not, write to the Free Software Foundation, Inc.,
+ * 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ */
+
+#include "filter.h"
+#include "filterlist.h"
+
+#include <CommHistory/Recipient>
+
+#include <abstractvoicecallprovider.h>
+
+using namespace VoiceCall;
+
+class Filter::Private
+{
+public:
+    Private()
+        : m_rejected(QString::fromLatin1("/sailfish/voicecall/filter/rejected-numbers"))
+        , m_ignored(QString::fromLatin1("/sailfish/voicecall/filter/ignored-numbers"))
+        , m_whitelist(QString::fromLatin1("/sailfish/voicecall/filter/whitelist"))
```

Comment:
Modern equivalent is `allowlist`. Unfortunately didn't know about this repo to review earlier until https://forum.sailfishos.org/t/release-notes-tampella-5-0-0-55/21801/231

Not sure if it's worth it to refactor retroactively, although with initial minimal usage, there will never be a better time.

Reply:
Hello, thank you for commenting. Do you mean more modern for inclusiveness and avoiding discrimination biais ? Ah, sorry for this I'm a bit too old to easily remember these kind of changes. I've no opinion if it's still early enough to change it or not.

Reply:
> Do you mean more modern for inclusiveness and avoiding discrimination biais?
> Ah, sorry for this I'm a bit too old to easily remember these kind of changes. I've no opinion if it's still early enough to change it or not.

Yes.

I'm also not a fan of spending energy on refactoring old code for such, but this is borderline new and exposes a user-facing API.

It's Jollyboys decision in the end, although a candidate PR could be filed regardless. I think the only semi-tough question is whether to support back-compatibility for those few people who already have entries stored in legacy-named keys :thinking: If a key can be easily renamed, that could simply be a release-based user instruction and problem solved?

Reply:
I wouldn't care about the modernity per se too much, whitelist has been there for centuries and will be around for some time more. Having that said, "allowed-numbers" type of naming would match the intention and the other configs nicer.

But is it worth adjusting at this point, not sure either. It's a lower level config which isn't directly visible, nor even used, in the UI.